### ISSUE TITLE
Lexer: Breaking change with one line object with function value returned

### ISSUE THREAD

**Issue by wdavidw** [jashkenas/coffeescript#4616]
Sorry if the title isnt clear, didn't really know how to express it, sometime one example worth a tousand words. 

Consider

```coffee
nikita()
.call header: 'h1', handler: (->)
.call header: 'h2', handler: (->)
```

In version 1.12.5

```js
(function() {
  nikita().call({
    header: 'h1',
    handler: (function() {})
  }).call({
    header: 'h2',
    handler: (function() {})
  });

}).call(this);
```

while in version 1.12.6

```js
(function() {
  nikita().call({
    header: 'h1',
    handler: (function() {}).call({
      header: 'h2',
      handler: (function() {})
    })
  });

}).call(this);
```

**Comment by helixbass** [comment]
@wdavidw this appears to have been fixed by #4598 so should return to the correct 1.12.5 compilation in the next `1.x` release

**Comment by wdavidw** [comment]
Great, didnt saw it, closing
